src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/FunctionProbes/FunctionProbesScenario.cs

```diff
@@ -0,0 +1,326 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.FunctionProbes;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using SampleMethods;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using static SampleMethods.StaticTestMethodSignatures;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTestApp.Scenarios.FunctionProbes
+{
+    internal static class FunctionProbesScenario
+    {
+        private delegate Task TestCaseAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, CancellationToken token);
+
+        public static CliCommand Command()
+        {
+            CliCommand command = new(TestAppScenarios.FunctionProbes.Name);
+            command.SetAction(ExecuteAsync);
+            return command;
+        }
+
+        public static Task<int> ExecuteAsync(ParseResult result, CancellationToken token)
+        {
+            Dictionary<string, TestCaseAsync> testCases = new()
```

Comment:
Rather than treating these as commands, I think it makes more sense to make them subscenarios e.g. https://github.com/dotnet/dotnet-monitor/blob/main/src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/ExceptionsScenario.cs#L15

These don't seem like they are for orchestration or used multiple times per test app process but represent distinct scenarios that you are trying to test.